nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/blob/BlobToPDF.java

```diff
@@ -48,12 +50,21 @@ public Blob run(DocumentModel doc) throws IOException {
         if (bh == null) {
             return null;
         }
-        return service.convertBlobToPDF(bh.getBlob());
+        String srcMT = bh.getBlob().getMimeType();
+        String destMT = MimetypeRegistry.PDF_MIMETYPE;
+        String extension = MimetypeRegistry.PDF_EXTENSION;
+        return service.convertBlob(service.getConverterName(srcMT, destMT), bh, extension, destMT);
     }
 
     @OperationMethod
     public Blob run(Blob blob) throws IOException {
-        return service.convertBlobToPDF(blob);
+        String srcMT = blob.getMimeType();
+        String destMT = MimetypeRegistry.PDF_MIMETYPE;
+        String extension = MimetypeRegistry.PDF_EXTENSION;
+        Blob result = service.convertBlob(service.getConverterName(srcMT, destMT), new SimpleBlobHolder(blob),
+                extension, destMT);
+        service.adjustBlobName(blob.getFilename(), result, extension, destMT);
```

Comment:
You shouldn't need to adjust blob name, it should be done in the `convertBlob` method.
As it you can remove  `adjustBlobName` from `ConfigurationService` interface.